### PR TITLE
Add support `cpu.cores` hardware requirement for mrack

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage
* [x] mention the version
* [x] include a release note

### CODE REVIEW

**Comment by happz** [comment]
@skycastlelily hello, please, rebase & add a simple unit test similar to https://github.com/teemtee/tmt/blob/main/tests/unit/provision/mrack/test_hw.py#L277? This PR is fairly simple, it should be easy to get it reviewed and merged quickly.

**Comment by skycastlelily** [comment]
Really sorry for the late reply, I'm just back from my national holiday.

> This PR is fairly simple, This PR is fairly simple, it should be easy to
get it reviewed and merged quickly.

Yes ,and updated^^

**Comment by skycastlelily** [comment]
updated^^

**Review by happz — APPROVED** [review]

**Comment by therazix** [comment]
> Oh, and one more thing: please, update specs/hardware/cpu.fmf, in a way similar to https://github.com/teemtee/tmt/pull/2733/files#diff-844c430661f020411c529315cabf0d4099728fc3226a2b448ceacfc6637fe550, and mention beaker gained support of new requirement.

@skycastlelily Could you please update the `specs/hardware/cpu.fmf` as well?

**Comment by skycastlelily** [comment]
> Could you please update the specs/hardware/cpu.fmf as well?

Thanks for your review,updated:)

**Comment by therazix** [comment]
Just a couple of things. Shouldn't the release note be for version 1.33 instead of 1.32? The same applies to the `versionchanged`.

Also, `cpu.cores` should probably be added to the `SUPPORTED_HARDWARE_CONSTRAINTS` list here:
https://github.com/teemtee/tmt/blob/22731b9761b62177de00ee7818c2f0ee4efcfe03/tmt/steps/provision/mrack.py#L51-L62

**Comment by skycastlelily** [comment]
Updated,thanks for your review again,and sorry for failing to find them by
myself:)

**Review by therazix — APPROVED** [review]

File: docs/releases.rst [hunk 1]
```diff
@@ -4,6 +4,17 @@
     Releases
 ======================
 
+tmt-1.33
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:ref:`Beaker</spec/plans/provision/beaker>` provision plugin gains
+support for :ref:`Cpu cores</spec/hardware/cpu>`HW requirement.
```
[the remark this answers is not in the document]
Comment (skycastlelily) [reply docs/releases.rst]: Updated^^

**Review by psss — APPROVED** [review]
Thanks for implementing this!

**Comment by skycastlelily** [comment]
My pleasure,thanks for your review:)

**Comment by psss** [comment]
Seems that `/tests/unit/with-system-packages/basic` now fails with:

    FAILED provision/mrack/test_hw.py::test_maximal_constraint - AssertionError: ...

@skycastlelily, could you please have a look?

**Comment by skycastlelily** [comment]
I forgot to update that testcase,should be good now:)

**Comment by psss** [comment]
Test failures are irrelevant, merged.

**Pull request merged**